taxi/tests/test_views.py

```diff
@@ -0,0 +1,64 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase, Client
+from django.urls import reverse
+
+from taxi.models import Manufacturer
+
+LITERARY_FORMAT_URL = reverse("taxi:manufacturer-list")
+
+
+class PublicManufacturerTests(TestCase):
+
+    def test_login_required(self):
+        res = self.client.get(LITERARY_FORMAT_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "password12345"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturer(self):
+        Manufacturer.objects.create(name="Tesla")
+        Manufacturer.objects.create(name="Mercedes-Benz")
```

Comment:
You can add country to Manufacturer object

Reply:
fixed too)